### PR TITLE
API v3 - use ssn as veteran identifier across endpoints

### PR DESCRIPTION
**Pull request by erluti**
Resolves https://github.com/department-of-veterans-affairs/caseflow/issues/13167

### Description
* documentation updated to use `ssn` as a parameter for identifying veterans
* higher_level_reviews#create updated so veteran object should include `ssn`
* contestable_issues#index updated so veteran in header is identified with `ssn`
* changes reflected in tests with some refactoring

### Acceptance Criteria
- [x] ssn is used as parameter to identify veteran

### CODE REVIEW

File: spec/requests/api/v3/decision_review/higher_level_reviews_controller_spec.rb [hunk 1]
```diff
@@ -9,21 +9,29 @@
     FeatureToggle.enable!(:api_v3)
 
     Timecop.freeze(post_ama_start_date)
+
+    # setup a rating
+    rating
```
Comment (pkarman) [inline spec/requests/api/v3/decision_review/higher_level_reviews_controller_spec.rb]: if you `let!(:rating)` with the bang, you don't need to call the fixture explicitly.

File: app/services/api/v3/decision_review/intake_processor.rb [hunk 2]
```diff
@@ -38,7 +38,7 @@ def uuid
   private
 
   def build_intake(user, form_type)
-    Intake.build(user: user, veteran_file_number: @params.file_number_or_ssn, form_type: form_type)
+    Intake.build(user: user, veteran_file_number: @params.veteran_file_number, form_type: form_type)
```
Comment (pkarman) [inline app/services/api/v3/decision_review/intake_processor.rb]: should that be `@params.veteran_ssn` instead?

File: spec/requests/api/v3/decision_review/higher_level_reviews_controller_spec.rb [hunk 3]
```diff
@@ -406,7 +406,7 @@ def get_higher_level_review # rubocop:disable Naming/AccessorMethodName
           request_issue_in_active_review = create(
             :request_issue,
             decision_date: Time.zone.today - 5.days,
-            decision_review: create(:higher_level_review, id: 10, veteran_file_number: veteran.file_number),
+            decision_review: create(:higher_level_review, id: 10, veteran_file_number: veteran.ssn),
```
Comment (edmangimelli) [inline spec/requests/api/v3/decision_review/higher_level_reviews_controller_spec.rb]: @pkarman
I'm not sure why Philip was using an ssn here, I think I might switch this back

File: spec/requests/api/v3/decision_review/higher_level_reviews_controller_spec.rb [hunk 4]
```diff
@@ -416,7 +416,7 @@ def get_higher_level_review # rubocop:disable Naming/AccessorMethodName
           ineligible_request_issue = create(
             :request_issue,
             decision_date: Time.zone.today - 3.days,
-            decision_review: create(:higher_level_review, id: 11, veteran_file_number: veteran.file_number),
+            decision_review: create(:higher_level_review, id: 11, veteran_file_number: veteran.ssn),
```
Comment (edmangimelli) [inline spec/requests/api/v3/decision_review/higher_level_reviews_controller_spec.rb]: @pkarman
(again) I'm not sure why Philip was using an ssn here, I think I might switch this back

File: app/services/api/v3/decision_review/intake_processor.rb (hunk 2 above)
Comment (edmangimelli) [reply app/services/api/v3/decision_review/intake_processor.rb]: Philip's line seems right to me (if he's wanting to use kwarg `veteran_file_number:  x`)

File: app/controllers/api/docs/v3/decision_reviews.yaml [hunk 5]
```diff
@@ -131,24 +131,23 @@ paths:
...
+                              pattern: '^[0-9]{9}$'
                               description: >-
-                                The veteran's file number or SSN.
+                                The veteran's SSN.
                                 Example: "123456789"
-                                Format: 8 or 9 digit characters.
-                                Max length: 9 characters.
-                                Corresponds to both "1. VETERAN'S SOCIAL SECURITY NUMBER" and "2. VA FILE NUMBER" on form 20-0996.
+                                Format: 9 digit characters.
+                                Corresponds to both "2. VETERAN'S SOCIAL SECURITY NUMBER" and "3. VA FILE NUMBER" on form 20-0996.
```
[the remark this answers is not in the document]
Comment (edmangimelli) [reply app/controllers/api/docs/v3/decision_reviews.yaml]: nice catch! i'll fix that

[the remark this answers is not in the document]
Comment (edmangimelli) [reply app/controllers/api/docs/v3/decision_reviews.yaml]: https://github.com/department-of-veterans-affairs/caseflow/pull/13297

**Pull request merged**